fv/wireguard_test.go

```diff
@@ -994,6 +985,30 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ WireGuard-Supported 3-node
 		}
 
 		cc = &connectivity.Checker{}
+
+		// Check felix Wireguards have handshaked.
+		for i, felix := range felixes {
+			var matchers []types.GomegaMatcher
```

Comment:
The preparation of `matchers` is unusual here.  I think it becomes `len(felixes)-1` copies of `BeNumerically(">", 0)` - right?  That feels surprisingly heavy, and it also feels strange that the matchers are constructed **inside** the Eventually func that you're testing.

Reply:
But I can see that, if it fails, it's nice to see the whole set of handshake numbers together.  So maybe just move the preparation of the matchers slice before the Eventually call - would that work?

Reply:
Thanks Neil, I updated to make the code clearer.